### DISCUSSION TITLE
Route path '/*.jpg' is triggered for any request

### DISCUSSION THREAD

**Discussion by ziimakc** [fastify/fastify#3347] [Q&A] [answered]
I want to catch all get `*.jpg` requests, but for some reason route with path `url: '/*.jpg'` is triggered for any possible route.

For example with get request by url `http://127.0.0.1:3001/smth.png` or `http://127.0.0.1:3001/dssdsd/sfsdfsdf` this route handler is triggered:

```
  instance.route({
    method: 'GET',
    url: '/*.jpg',
    handler: async (req) => {
      console.error({ routerPath: req.routerPath }); // { routerPath: '/*.jpg' }
      console.error({ url: req.url }); // { url: '/smth.png' }

      return 2;
    },
  });
```

Is there some wildcard problem I'm missing?

**Comment by zekth** [comment, marked as answer]
In your case you want to use a regex and not a glob pattern.
```js
instance.route({
  method: 'GET',
  url: '/:file(^\\d+).jpg',
  handler: async (req) => {
    console.error({ routerPath: req.routerPath }); // { routerPath: '/*.jpg' }
    console.error({ url: req.url }); // { url: '/smth.png' }

    return 2;
  },
});
```

But i definitely would not recommend this as regex are really expensive. It's better to check if the path is a file in the handler instead.

See: https://www.fastify.io/docs/latest/Routes/#url-building

**Reply by ziimakc** [reply]
@zekth so if I use `*` everything after doesn't matter?

**Reply by delvedor** [reply]
It should matter, but currently there is a bug.
See https://github.com/delvedor/find-my-way/issues/202
